pkg/cloud/vsphere/services/govmomi/create.go

```diff
@@ -39,8 +39,8 @@ const (
 	// hostnameLookup resolves via cloud init and uses cloud provider's metadata service to lookup its own hostname.
 	hostnameLookup = "{{ ds.meta_data.hostname }}"
 
-	// containerdSocket is the path to containerd socket.
-	containerdSocket = "/var/run/containerd/containerd.sock"
+	// criSocket is the path to the CRI socket to connect. If empty kubeadm will try to auto-detect this value.
+	criSocket = ""
```

Comment:
@detiber would love your thoughts on this - it would allow machine images that prefer docker as the container runtime.

Reply:
Longer term, my thoughts here is that we could add fields like this to a centralized location in cluster-api, which would then be able to impact both the bootstrapping behavior (setting the right kubeadm configuration) and also the right image lookup behavior, so that both configurations would match with less likelihood of a user shooting themselves in the foot.
/cc @ncdc @vincepri

Reply:
Also see https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/863/files for how we adjusted CAPA to default to containerd but allow the user to override.